mod-pubsub-client/src/test/java/org/folio/util/pubsub/AbstractRestTest.java

```diff
@@ -0,0 +1,165 @@
+package org.folio.util.pubsub;
+
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import net.mguenther.kafka.junit.EmbeddedKafkaCluster;
+import net.mguenther.kafka.junit.TopicConfig;
+import org.folio.rest.RestVerticle;
+import org.folio.rest.client.TenantClient;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.tools.utils.NetworkUtils;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+
+import static java.lang.String.format;
+import static net.mguenther.kafka.junit.EmbeddedKafkaCluster.provisionWith;
+import static net.mguenther.kafka.junit.EmbeddedKafkaClusterConfig.useDefaults;
+import static org.folio.rest.RestVerticle.OKAPI_HEADER_TENANT;
+
+public abstract class AbstractRestTest {
+
+  protected static final String TENANT_ID = "diku";
+  protected static final String TOKEN = "token";
+  private static final String HTTP_PORT = "http.port";
+  private static final String DELETE_ALL_SQL = "DELETE FROM pubsub_config.%s";
+  private static final String EVENT_DESCRIPTOR_TABLE = "event_descriptor";
+  private static final String MESSAGING_MODULE_TABLE = "messaging_module";
+  private static final String AUDIT_MESSAGE_PAYLOAD_TABLE = "audit_message_payload";
+  private static final String AUDIT_MESSAGE_TABLE = "audit_message";
+
+  private static final String KAFKA_HOST = "KAFKA_HOST";
+  private static final String KAFKA_PORT = "KAFKA_PORT";
+
+  static RequestSpecification spec;
+  private static int port;
+  private static String useExternalDatabase;
+  private static Vertx vertx;
+  protected static String okapiUrl;
+
+  @ClassRule
+  public static EmbeddedKafkaCluster cluster = provisionWith(useDefaults());
+
+  @BeforeClass
+  public static void setUpClass(final TestContext context) throws Exception {
+    vertx = Vertx.vertx();
+    runDatabase();
+    cluster.createTopic(TopicConfig.forTopic("pub-sub.diku.record_created").build());
```

Comment:
Please move 57-60 lines into a separate method, just to have 
runDatabase();
createTopic();
deployVerticle(context);